org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/xsd/contentmodel/CMXSDAttributeDeclaration.java

```diff
@@ -90,6 +105,67 @@ private XSObjectList getAnnotations() {
 		return null;
 	}
 
+	/**
+	 * Returns list of xs:annotation from the element declaration or type
+	 * declaration.
+	 * 
+	 * Indicated by:
+	 * https://msdn.microsoft.com/en-us/library/ms256143(v=vs.110).aspx
+	 * xs:attribute tags have content of either an xs:annotation or xs:simpleType
+	 * 
+	 * @return list of xs:annotation from the element declaration or type
+	 *         declaration.
+	 */
+	private XSObjectList getValueAnnotations() {
+		// Try get xs:annotation from the element declaration
+		XSAttributeDeclaration attributeDeclaration = getAttrDeclaration();
+		XSSimpleTypeDefinition simpleTypeDefinition = attributeDeclaration.getTypeDefinition();
+		XSSimpleTypeDecl simpleTypeDecl;
+		
+	
+		XSObjectList annotation = null; // The XSD tag that holds the documentation tag
+
+		if(simpleTypeDefinition != null && simpleTypeDefinition instanceof XSSimpleTypeDecl) {
+			simpleTypeDecl = (XSSimpleTypeDecl) simpleTypeDefinition;
+			XSObjectList tempAnnotation = simpleTypeDecl.getMultiValueFacets();
+			XSMultiValueFacet zz = (XSMultiValueFacet) tempAnnotation.get(0);
+			tempAnnotation = zz.getAnnotations();
+			Object[] annotationArray = tempAnnotation.toArray();

```

Comment:
do you really get back arrays containing null-elements?

Reply:
yes